lib/resource-mapper.js

```diff
@@ -68,11 +70,9 @@ class ResourceMapper {
       // Read all files in the corresponding folder
       const filename = fullPath.substr(fullPath.lastIndexOf('/') + 1)
       const folder = fullPath.substr(0, fullPath.length - filename.length)
-      const files = await this._readdir(folder)
 
       // Find a file with the same name (minus the dollar extension)
-      let match = !searchIndex ? '' : (files.find(f => this._removeDollarExtension(f) === filename ||
-        (isIndex && f.startsWith(this._indexFilename + '.'))))
+      let match = searchIndex ? await this._getMatchingFile(folder, filename, isIndex) : ''
```

Comment:
I'm not sure if I would move this specific bit to a new function. Is there potential for reuse?

Reply:
My thinking was to make it more readable by giving it a name. Don't know of any potential reuse, no.